doc/source/conf.py

```diff
@@ -332,27 +333,86 @@ def prepare_jinja_env(jinja_env) -> None:
 }
 
 
-def copy_download_files_to_source_dir(app: sphinx.application.Sphinx) -> None:
+def archive_examples(app: sphinx.application.Sphinx) -> None:
     """
-    Copy zipped example files to target directory at build time
+    Create a zip archive for each listed example included in the examples folder.
 
     Parameters
     ----------
     app : sphinx.application.Sphinx
         Sphinx application instance containing the all the doc build configuration.
 
     """
-
-    # archive_examples.py in the root directory is being run before doc build (via tox)
-    # we simply need to copy the into the _download target defined in the .rst source files.
-    SOURCE_DIR = pathlib.Path(app.srcdir)
-    ZIPPED_FILES_DIR = SOURCE_DIR.parent.parent / "build"
-    DOWNLOAD_FILES_DIR = SOURCE_DIR.parent / "_build" / "html" / "_downloads"
-    DOWNLOAD_FILES_DIR.mkdir(exist_ok=True)
-
-    for file_path in ZIPPED_FILES_DIR.glob("*"):
+    EXAMPLES = {
```

Comment:
why using all caps for variables like examples, source_dir etc?

Reply:
This is because those more or less do not change, like constants. 
If you prefer, I can move examples to the top of the file and pass it as an argument to the function instead.

Reply:
`SOURCE_DIR` should just be a lower case variable, it depends on the input `app`, it's not a constant.
Regarding `EXAMPLES`, either you move it outside of the function as a constant or you make it lower case inside the function. Either is fine with me.

Reply:
Thanks! Implemented your suggestions.